protocols/imap/src/main/java/org/apache/james/imap/api/process/SelectedMailbox.java

```diff
@@ -93,6 +95,11 @@
      */
     MailboxId getMailboxId();
 
+    /**
+     * @return The associated MessageManager without further dataStore reads
+     */
+    MessageManager getMessageManager() throws MailboxNotFoundException;
```

Comment:
The problem with that is: MessageManager is not exactly stateless, it contains _at least_ one entity, `Mailbox`.
This entity has itself at least two fields that can change in database over time: acl and uidValidity.
You then can't rely on keeping this MessageManager as a field

Reply:
Controversial commit dropped ;-)